components/action-mgt/org.wso2.carbon.identity.action.management/src/test/java/org/wso2/carbon/identity/action/management/dao/ActionManagementDAOImplTest.java

```diff
@@ -0,0 +1,311 @@
+        identityUtil.close();
+    }
+
+    @AfterClass
+    public void wrapUp() throws Exception {
+
+        closeH2Database();
+    }
+
+    @Test(priority = 1)
+    public void testAddAction() throws ActionMgtException, SQLException {
+
+        Action action = buildMockAction(false,false);
+        Action result = daoImpl.addAction(PRE_ISSUE_ACCESS_TOKEN, action.getId(), action, TENANT_ID);
+        Assert.assertEquals(action.getId(), result.getId());
+        Assert.assertEquals(action.getName(), result.getName());
+        Assert.assertEquals(action.getDescription(), result.getDescription());
+        Assert.assertEquals(action.getType(), result.getType());
+        Assert.assertEquals(action.getStatus(), result.getStatus());
+        Assert.assertEquals(action.getEndpoint().getUri(), result.getEndpoint().getUri());
+        Assert.assertEquals(action.getEndpoint().getAuthentication().getType(), result.getEndpoint().getAuthentication().getType());
+    }
+
+    @Test(priority = 2)
+    public void testGetActionsByActionType() throws ActionMgtException, SQLException {
+
+        Action action = buildMockAction(false,false);
+        Assert.assertEquals(1, daoImpl.getActionsByActionType(PRE_ISSUE_ACCESS_TOKEN, TENANT_ID).size());
+        loadMockDatabaseUtil();
+        daoImpl.addAction(PRE_ISSUE_ACCESS_TOKEN, action.getId(), action, TENANT_ID);
+        loadMockDatabaseUtil();
+        Assert.assertEquals(2, daoImpl.getActionsByActionType(PRE_ISSUE_ACCESS_TOKEN, TENANT_ID).size());
+    }
+
+    @Test(priority = 3)
+    public void testDeleteAction() throws ActionMgtException, SQLException {
+
+        Action action = buildMockAction(false,true);
+        Assert.assertEquals(3, daoImpl.getActionsByActionType(PRE_ISSUE_ACCESS_TOKEN, TENANT_ID).size());
+        loadMockDatabaseUtil();
+        daoImpl.deleteAction(PRE_ISSUE_ACCESS_TOKEN, action.getId(), action, TENANT_ID);
+        loadMockDatabaseUtil();
+        Assert.assertEquals(2, daoImpl.getActionsByActionType(PRE_ISSUE_ACCESS_TOKEN, TENANT_ID).size());
+    }
+
+    @Test(priority = 4)
+    public void testAddActionWithoutOptional() throws ActionMgtException, SQLException {
+
+        Action action = buildMockAction(true,false);
+        Action result = daoImpl.addAction(PRE_ISSUE_ACCESS_TOKEN, action.getId(), action, TENANT_ID);
+        Assert.assertEquals(action.getId(), result.getId());
+        Assert.assertEquals(action.getName(), result.getName());
+        Assert.assertEquals(action.getType(), result.getType());
+        Assert.assertEquals(action.getStatus(), result.getStatus());
+        Assert.assertEquals(action.getEndpoint().getUri(), result.getEndpoint().getUri());
+        Assert.assertEquals(action.getEndpoint().getAuthentication().getType(), result.getEndpoint().getAuthentication().getType());
+    }
+
+    @Test(priority = 5)
+    public void testUpdateAction() throws ActionMgtException, SQLException {
+
+        Action action = buildMockAction(true,true);
```

Comment:
We don't need to create another action. Lets use the action created in above method and update that action. 
Store the id of the action in a class level variable and use the id here

Reply:
Okay I hold the primary action getting used through out the test class as a class level variable.